server/services/datacommons.py

```diff
@@ -263,6 +263,55 @@ def v2node(nodes, prop):
   })
 
 
+def _merge_paged_response(result, paged_response):
+  for dcid in paged_response.get('data', []):
+    # Initialize dcid in data even when no arcs or properties are returned
+    result.setdefault('data', {}).setdefault(dcid, {})
+
+    for prop in paged_response['data'][dcid].get('arcs', {}):
+      result['data'][dcid].setdefault('arcs', {}).setdefault(
+          prop, {}).setdefault('nodes', []).extend(
+              paged_response['data'][dcid]['arcs'][prop]['nodes'])
+
+    if 'properties' in paged_response['data'][dcid]:
+      result['data'][dcid].setdefault('properties', []).extend(
+          paged_response['data'][dcid].get('properties', []))
+
+  result['nextToken'] = paged_response.get('nextToken', '')
+  if not result['nextToken']:
+    del result['nextToken']
+
+
+def v2node_paginated(nodes, prop, max_pages=1):
+  """Wrapper to call V2 Node REST API.
+
+  Args:
+      nodes: A list of node dcids.
+      prop: The property to query for.
+      max_pages: The maximum number of pages to fetch. If None, v2node is
+        queried until nextToken is not in the response.
+  """
+  fetched_pages = 0
+  result = {}
+  next_token = ''
+  while True:
+    url = get_service_url('/v2/node')
+    response = post(url, {
+        'nodes': sorted(nodes),
+        'property': prop,
+        'nextToken': next_token
+    })
+    if max_pages == 1:
```

Comment:
instead of specifically returning response when max_pages == 1, maybe _merge_paged_response() can just return response if result is empty?

Reply:
Ah sorry, this was from before when I was directly editing the v2node function. Now that I'm just creating a new one, I can delete this condition.

Thanks for calling it out!!